### PR TITLE
docs: add "Cloud Connection API and Usage" section 

### PR DESCRIPTION
**Pull request by mattdibi**
Added "Cloud Connection API and Usage" section. This is a simple port of the one available on the `gh-pages-backup` branch with minor style changes.

### CODE REVIEW

**Comment by marcellorinaldo** [comment]
@mattdibi is this a work in progress?

**Comment by mattdibi** [comment]
> @mattdibi is this a work in progress?

No, why?

File: docs/cloud-api/app-dev-guide.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+The Kura [ExamplePublisher](https://github.com/eclipse/kura/tree/develop/kura/examples/org.eclipse.kura.example.publisher) will be used as a reference.
+
+The application should bind itself to a `CloudPublisher` or `CloudSubscriber` instance, this can be done in different ways, such as using OSGi `ServiceTracker`s or by leveraging the Declarative Service layer.
+
+The recommended way to perform this operation is choosing the latter and allowing the user to customize the service references through component configuration.
+
+If the component metatype and definition are structured as described below, the Kura Web UI will show a dedicated widget in component configuration that helps the user to pick compatible `CloudPublisher` or `CloudSubscriber` instances.
+
+1. **Write component definition**
```
Comment (marcellorinaldo) [inline docs/cloud-api/app-dev-guide.md:13]: Enumeration will not work in this list if you don't indent the rest. Maybe those can be subsections.

File: docs/cloud-api/built-in-cloud.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+- Adds application topic prefixes to allow for a single remote server connection to be shared across applications.
+
+- Defines a payload data model and provides default encoding/decoding serializers.
+
+- Publishes life-cycle messages when the device and applications start and stop.
+
+To use this service, select the **CloudServices** option located in the **System** area and select the **CloudService** tab as shown in the screen capture below.
+
+![](images/cloudService.png){ style="border-radius: 7px;"}
```
Comment (marcellorinaldo) [inline docs/cloud-api/built-in-cloud.md:27]: Some old screenshots, but I think we can keep them.

File: docs/cloud-api/overview.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+```java
+public interface RequestHandler {
+
+    public KuraMessage doGet(RequestHandlerContext context, KuraMessage reqMessage) throws KuraException;
+
+    public KuraMessage doPut(RequestHandlerContext context, KuraMessage reqMessage) throws KuraException;
+
+    public KuraMessage doPost(RequestHandlerContext context, KuraMessage reqMessage) throws KuraException;
+
+    public  KuraMessage doDel(RequestHandlerContext context, KuraMessage reqMessage) throws KuraException;
```
Comment (marcellorinaldo) [inline docs/cloud-api/overview.md]: There is an extra space before KuraMessage

File: docs/cloud-api/user-guide.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+The involved steps are the following
+
+1. [Instantiation and configuration of the Cloud Connection](#creating-a-new-cloud-connection).
+2. [Instantiation and configuration of a Publisher](#creating-and-configuring-a-new-publisher).
+3. [Binding an application to the Publisher](#binding-an-application-to-a-publisher).
+
+## Creating a new Cloud Connection
+
+1. Open the Cloud Connections section of the Web UI:
```
Comment (marcellorinaldo) [inline docs/cloud-api/user-guide.md:13]: Need to indent the image to have the enumeration working for the list. Please apply to the rest of the file

**Review by marcellorinaldo — CHANGES_REQUESTED** [review]

**Comment by mattdibi** [comment]
@marcellorinaldo should be fixed now

**Review by marcellorinaldo — APPROVED** [review]

**Pull request merged**